resources/todomvc/architecture-examples/preact/src/app/app.js

```diff
@@ -0,0 +1,69 @@
+// eslint-disable-next-line no-unused-vars
+import { h } from "preact";
+import { useEffect, useState } from "preact/hooks";
+
+import TodoModel from "./model";
+import TodoHeader from "./header";
+import TodoMain from "./main.js";
+import TodoFooter from "./footer";
+import { FILTERS } from "./utils";
+
+const getRoute = () => {
+    let route = String(location.hash || "")
+        .split("/")
+        .pop();
+
+    // prettier-ignore
+    if (!FILTERS[route])
+        route = "all";
+
+    return route;
+};
+
+export default function App() {
+    const [, setUdpatedAt] = useState(Date.now());
+    const [route, setRoute] = useState("all");
+
+    function update() {
+        setUdpatedAt(Date.now());
```

Comment:
since model updates a local array and doesn't use state, the update function that gets called after each model change simply triggers a re-render (**similar to the previous version**).

Reply:
Here is another solution to force an update: https://reactjs.org/docs/hooks-faq.html#is-there-something-like-forceupdate
It's more elegant because it doesn't need to call `Date.now()`, which can by itself be slow on some systems.

Alternatively, you could use `useState` with an integer state and call `setValue(val => val + 1)` but I find this less explicit.

Reply:
added a note